pkg/lua/contrib/lua_run.c

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @ingroup  pkg_lua
+ * @ingroup  pkg
+ * @ingroup  sys
+ * @{
+ * @file
+ *
+ * @brief   Convenience functions for running Lua code.
+ * @author  Juan Carrano <j.carrano@fu-berlin.de>
+ *
+ */
+
+#define LUA_LIB
+
+#include "lprefix.h"
+
+#include <setjmp.h>
+
+#include "kernel_defines.h"
+#include "tlsf.h"
+
+#include "lua.h"
+#include "lualib.h"
+#include "lauxlib.h"
+
+#include "lua_run.h"
+#include "lua_loadlib.h"
+
+const char * luaR_str_errors[] = {
+    "No errors",
+    "Error setting up the interpreter",
+    "Error while loading a builtin library",
+    "Cannot find the specified module",
+    "Compilation / syntax error",
+    "Unprotected error (uncaught exception)",
+    "Out of memory",
+    "Internal interpreter error",
+    "Unknown error"
+};
+
+/* The lua docs state the behavior in these cases:
+ *
+ *  1.              ptr=?, size=0    -> free(ptr)
+        therefore   ptr=NULL, size=0 -> NOP
+ *  2.               ptr=? , size!=0  -> realloc(ptr, size)
+ *
+ * The  TLSF code for realloc says:
+ *  / * Zero-size requests are treated as free. * /
+ *  if (ptr && size == 0)
+ *  {
+ *      tlsf_free(tlsf, ptr);
+ *  }
+ *  / * Requests with NULL pointers are treated as malloc. * /
+ *  else if (!ptr)
+ *  {
+ *      p = tlsf_malloc(tlsf, size);
+ *  }
+ *
+ * Therefore it is safe to use tlsf_realloc here.
+ */
+static void *lua_tlsf_alloc(void *ud, void *ptr, size_t osize, size_t nsize)
+{
+    tlsf_t tlsf = ud;
+    (void)osize;
+
+    return tlsf_realloc(tlsf, ptr, nsize);
+}
+
+LUALIB_API lua_State *luaR_newstate(void *memory, size_t mem_size,
```

Comment:
@jcarrano @danpetry I'm not sure about these luaR_xxx functions. Are these Lua or RIOT specific? According to our [Coding Conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions), RIOT code should be always snake_case( It's OK to call CamelCase functions though if they are defined in the original pkg)


What do you think?

Reply:
(it's also OK to write CamelCase when it's the implementation of a function required by the pkg, but couldn't find any luaR functions reference on internet)

Reply:
There is no "nice" solution. I chose those names because those are replacements for `luaL_newstate`, `lual_openlibs`, etc, and the convention in lua is to have lua_xxxx for core functions and luaY_xxx for other functions.

Reply:
I tend to think this code should follow RIOT standards, because it's an adoption for RIOT.
Maybe using wrappers as seen [here](https://github.com/RIOT-OS/RIOT/blob/master/pkg/semtech-loramac/include/semtech_loramac.h#L162) can do the trick.

Note is always possible to directly invoke pkg functions with their own style, but this code will live in Doxygen and strictly attached to the RIOT side

Reply:
There is still the rest of the lua api, named after lua conventions. Then I would have mixed conventions in the same code (lua_riot_newstate returning a lua_State).

Reply:
My opinion is that it would be best to follow the RIOT conventions as far as possible. If the API is RIOT programmer-facing, then it would avoid the case where e.g. there are a bunch of different packages with their own coding style and the programmer sees a range of different styles. I think leaving in stuff which **has** to be left in is acceptable (and obv necessary) but better than having a real mix of styles

Reply:
I strongly agree with @jia200x and @danpetry here to follow RIOT coding conventions. mixing conventions is something that is often caused by external software. I would not wrap everything, but having at least our own contributed functions in our own convention also makes it easier for seasoned RIOT users as they know what to expect and prevents confusion with new users

Rewording our own functions to something like `lua_riot_funcname` is IMHO clear enough to indicate that it is our version of `funcname`.

(I think I accidentally rephrased @danpetry's arguments)

Reply:
This has been changed